### PR TITLE
feat: Add the RemoveMicrosoftApplicationInsightsLoggerProvider extension to all Serilog infrastructure code (🚧)

### LINKED ISSUE #604: Use Observability's `RemoveMicrosoftApplicationInsightsLoggerProvider` extension in Azure Functions project templates
**Is your feature request related to a problem? Please describe.**
When looking at improving the Application Insights telemetry in Azure Functions, we saw that we should remove the default Microsoft's telemetry as it conflicts with Arcus' telemetry. Therefore, the Observability library has created an `.RemoveMicrosoftApplicationInsightsLoggerProvider` extension that we can use when setting up our Serilog infrastructure.

**Describe the solution you'd like**
Add the `RemoveMicrosoftApplicationInsightsLoggerProvider` extension to all Serilog infrastructure code in all Azure Functions project templates.

**Additional context**
Relates to https://github.com/arcus-azure/arcus.observability/issues/364

### PR DESCRIPTION
**Pull request by vtermaat**
Add the RemoveMicrosoftApplicationInsightsLoggerProvider extension to all Serilog infrastructure code.
Closes #604

### CODE REVIEW

File: src/Arcus.Templates.AzureFunctions.Databricks.JobMetrics/Startup.cs [hunk 1]
```diff
@@ -41,8 +42,9 @@ public override void Configure(IFunctionsHostBuilder builder)
             LoggerConfiguration logConfig = CreateLoggerConfiguration(builder);
             builder.Services.AddLogging(logging =>
             {
-                logging.AddSerilog(logConfig.CreateLogger(), dispose: true);
+                logging.RemoveMicrosoftApplicationInsightsLoggerProvider().AddSerilog(logConfig.CreateLogger(), dispose: true);
             });
+
```
Comment (stijnmoreels) [inline src/Arcus.Templates.AzureFunctions.Databricks.JobMetrics/Startup.cs:47]:
```suggestion
                logging.RemoveMicrosoftApplicationInsightsLoggerProvider()
                       .AddSerilog(logConfig.CreateLogger(), dispose: true);
            });
```

Let's format this for readability! 👍 (others as well).

**Review by stijnmoreels — CHANGES_REQUESTED** [review]
This is great, Vincent! 🥇 Thx or this. Let's wait until we have the observability fully released so we can update this PR with the next minor. But already great job in helping us here!

File: src/Arcus.Templates.AzureFunctions.Databricks.JobMetrics/Startup.cs (hunk 1 above)
Comment (vtermaat) [reply src/Arcus.Templates.AzureFunctions.Databricks.JobMetrics/Startup.cs]: > Let's format this for readability! 👍 (others as well).

done 👍 https://github.com/arcus-azure/arcus.templates/pull/607/commits/1ca3dbb8b640da9257d097b0b6ea0aaceef8ef38

File: src/Arcus.Templates.AzureFunctions.Databricks.JobMetrics/Startup.cs [hunk 2]
```diff
@@ -41,8 +42,10 @@ public override void Configure(IFunctionsHostBuilder builder)
             LoggerConfiguration logConfig = CreateLoggerConfiguration(builder);
             builder.Services.AddLogging(logging =>
             {
-                logging.AddSerilog(logConfig.CreateLogger(), dispose: true);
+                logging.RemoveMicrosoftApplicationInsightsLoggerProvider()
+                    .AddSerilog(logConfig.CreateLogger(), dispose: true);
             });
+
```
Comment (stijnmoreels) [inline src/Arcus.Templates.AzureFunctions.Databricks.JobMetrics/Startup.cs:48]:
```suggestion
                logging.RemoveMicrosoftApplicationInsightsLoggerProvider()
                       .AddSerilog(logConfig.CreateLogger(), dispose: true);
            });
```

File: src/Arcus.Templates.AzureFunctions.Http/Startup.cs [hunk 3]
```diff
@@ -51,7 +52,8 @@ public override void Configure(IFunctionsHostBuilder builder)
             LoggerConfiguration logConfig = CreateLoggerConfiguration(builder);
             builder.Services.AddLogging(logging =>
             {
-                logging.AddSerilog(logConfig.CreateLogger(), dispose: true);
+                logging.RemoveMicrosoftApplicationInsightsLoggerProvider()
+                    .AddSerilog(logConfig.CreateLogger(), dispose: true);
```
Comment (stijnmoreels) [inline src/Arcus.Templates.AzureFunctions.Http/Startup.cs:56]:
```suggestion
                logging.RemoveMicrosoftApplicationInsightsLoggerProvider()
                       .AddSerilog(logConfig.CreateLogger(), dispose: true);
```

File: src/Arcus.Templates.AzureFunctions.ServiceBus.Queue/Startup.cs [hunk 4]
```diff
@@ -46,7 +47,8 @@ public override void Configure(IFunctionsHostBuilder builder)
             LoggerConfiguration logConfig = CreateLoggerConfiguration(builder);
             builder.Services.AddLogging(logging =>
             {
-                logging.AddSerilog(logConfig.CreateLogger(), dispose: true);
+                logging.RemoveMicrosoftApplicationInsightsLoggerProvider()
+                    .AddSerilog(logConfig.CreateLogger(), dispose: true);
```
Comment (stijnmoreels) [inline src/Arcus.Templates.AzureFunctions.ServiceBus.Queue/Startup.cs:51]:
```suggestion
                logging.RemoveMicrosoftApplicationInsightsLoggerProvider()
                       .AddSerilog(logConfig.CreateLogger(), dispose: true);
```

File: src/Arcus.Templates.AzureFunctions.ServiceBus.Topic/Startup.cs [hunk 5]
```diff
@@ -46,7 +47,8 @@ public override void Configure(IFunctionsHostBuilder builder)
             LoggerConfiguration logConfig = CreateLoggerConfiguration(builder);
             builder.Services.AddLogging(logging =>
             {
-                logging.AddSerilog(logConfig.CreateLogger(), dispose: true);
+                logging.RemoveMicrosoftApplicationInsightsLoggerProvider()
+                    .AddSerilog(logConfig.CreateLogger(), dispose: true);
```
Comment (stijnmoreels) [inline src/Arcus.Templates.AzureFunctions.ServiceBus.Topic/Startup.cs:51]:
```suggestion
                logging.RemoveMicrosoftApplicationInsightsLoggerProvider()
                       .AddSerilog(logConfig.CreateLogger(), dispose: true);
```

**Review by stijnmoreels — CHANGES_REQUESTED** [review]
Great stuff! 👍 Thx for fixing.
Let's see if we can release observability this week.

**Pull request closed without merging**